### PR TITLE
Add `plugin-modernizer-tool` permission and CD

### PR DESCRIPTION
**Pull request by jonesbusy**
# Link to GitHub repository

This is the tool for the GSoC 2024 OpenRewrite

https://github.com/jenkinsci/plugin-modernizer-tool/pull/8

Note: I've temporary disabled action to avoid the CD to run. We are in a very early stage of this component and I don't think it's worth publishing any releases before some weeks/month

Note: This previous PR also add the Jenkinsfile even no job exists yet on ci.jenkins.io. I will open a support ticket for it

# When modifying release permission

List the GitHub usernames of the users who should have commit permissions below:
- `@berviantoleo` --> To be added later due to missing login
- `@gounthar`
- `@iamrajiv` --> To be added later due to missing login
- `@jonesbusy`

This is needed in order to cut releases of the plugin or component.

If you are modifying the release permission of your plugin or component, fill out the following checklist:

```[tasklist]
### Release permission checklist (for submitters)
- [ ] The usernames of the users added to the "developers" section in the .yml file are the same the users use to log in to [accounts.jenkins.io](https://accounts.jenkins.io/).
- [ ] All users added have logged in to [Artifactory](https://repo.jenkins-ci.org/) and [Jira](https://issues.jenkins.io/) once.
- [ ] I have mentioned an [existing team member](https://github.com/orgs/jenkinsci/teams) of the plugin or component team to approve this request.
```

## When enabling automated releases (cd: true)

Follow the [documentation](https://www.jenkins.io/doc/developer/publishing/releasing-cd/) to ensure, your pull request is set up properly. Don't merge it yet.  
In case of changes requested by the hosting team, an open PR facilitates future reviews, without derailing work across multiple PRs.

### Link to the PR enabling CD in your plugin

https://github.com/jenkinsci/plugin-modernizer-tool/pull/8

```[tasklist]
### CD checklist (for submitters)
- [x] I have provided a link to the pull request in my plugin, which enables CD according to the documentation. 
```

```[tasklist]
### Reviewer checklist
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.
```
There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it.

### CODE REVIEW

**Comment by jonesbusy** [comment]
FYI @berviantoleo @gounthar @iamrajiv

**Comment by NotMyFault** [comment]
Some logins are missing still, but I think you can handle that on your own :)

**Review by iamrajiv — APPROVED** [review]

File: permissions/component-plugin-modernizer-tool.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+name: "plugin-modernizer-tool "
```
Comment (timja) [inline permissions/component-plugin-modernizer-tool.yml:2]:
```suggestion
name: "plugin-modernizer-tool"
```

**Review by timja — APPROVED** [review]

**Review by gounthar — APPROVED** [review]

**Comment by jonesbusy** [comment]
I've excluded missing login(s) or unknown users for now until login are fixed

**Pull request merged**